### PR TITLE
No Buffer

### PR DESCRIPTION
**Pull request by RikkiGibson**
Follow up to Azure/ms-rest-js#124

### CODE REVIEW

File: test/azure/generated/AzureCompositeModelClient/operations/primitive.ts [hunk 1]
```diff
@@ -2068,8 +2068,8 @@ export class Primitive {
     let field = (options && options.field !== undefined) ? options.field : undefined;
     // Validate
     try {
-      if (field && !Buffer.isBuffer(field)) {
-        throw new Error('field must be of type buffer.');
+      if (field && !(field instanceof Uint8Array)) {
+        throw new Error('field must be of type uint8array.');
```
Comment (daschult) [inline test/azure/generated/AzureCompositeModelClient/operations/primitive.ts]: Should the casing be the same here as in the previous line (Uint8Array)?

File: test/vanilla/complexTypesTests.ts [hunk 2]
```diff
@@ -379,22 +394,22 @@ describe('typescript', function () {
...
+        testClient.polymorphism.putValidMissingRequired(getBadfish(), function (error, result) {
           should.exist(error);
           error.message.should.containEql('birthday');
           error.message.should.containEql('cannot be null or undefined');
           done();
         });
       });
 
-      var rawSalmon: AutoRestComplexTestServiceModels.SmartSalmon = {
+      var getRawSalmon: () => AutoRestComplexTestServiceModels.SmartSalmon = () => ({
```
Comment (daschult) [inline test/vanilla/complexTypesTests.ts]: Why did this change?

File: test/vanilla/acceptanceTests.ts [hunk 3]
```diff
@@ -489,7 +510,13 @@ describe('typescript', function () {
       it('should get invalid byte value', function (done) {
         testClient.byteModel.getInvalid(function (error, result) {
           should.not.exist(error);
-          assert.deepEqual(result, new Buffer(':::SWAGGER::::', 'base64'));
+          // Base64 decoding of the string ':::SWAGGER::::'
+          const expected = new Uint8Array([73, 96, 6, 24, 68]);
```
Comment (RikkiGibson) [inline test/vanilla/acceptanceTests.ts]: This smells funny--will want to check on what the real intent of this test is and maybe adjust accordingly

File: test/vanilla/complexTypesTests.ts (hunk 2 above)
Comment (RikkiGibson) [reply test/vanilla/complexTypesTests.ts]: I wanted to be able to delete properties from this without worrying about the state of it in subsequent usages (e.g. doing a put after the checking the result of the get)

File: test/azure/generated/AzureCompositeModelClient/operations/primitive.ts (hunk 1 above)
Comment (RikkiGibson) [reply test/azure/generated/AzureCompositeModelClient/operations/primitive.ts]: That seems best, but we probably have a .ToLower() in this generator step for some reason.

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
### Size decreased by 0.50% (-10.01 KB)

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
### Size decreased by 0.99% (-19.82 KB)

**Pull request merged**